pkg/controllers/networking/pod_ip_cache.go

```diff
@@ -0,0 +1,161 @@
+/*
+ Copyright 2021 The Hybridnet Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package networking
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/alibaba/hybridnet/pkg/constants"
+
+	"github.com/alibaba/hybridnet/pkg/controllers/utils"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PodIPCache interface {
+	Record(podUID types.UID, podName, namespace string, ipInstanceNames []string)
+	Release(ipInstanceName, namespace string) error
+	Get(podName, namespace string) (bool, types.UID, []string)
+}
+
+type podAllocatedInfo struct {
+	podUID          types.UID
+	ipInstanceNames []string
```

Comment:
how about using a set instead of slice?

Reply:
I don't think it's necessary right now, maybe we will do it later.